enclave_build/src/image_manager.rs

```diff
@@ -0,0 +1,144 @@
+                return Self {
+                    image_name,
+                    storage: None,
+                };
+            }
+        };
+
+        // Try to create/read the storage
+        let storage = match OciStorage::new(&root_path) {
+            Ok(manager) => Some(manager),
+            Err(err) => {
+                // If the storage could not be created, log the error
+                eprintln!("{:?}", err);
+                None
+            }
+        };
+
+        Self {
+            image_name,
+            storage,
+        }
+    }
+
+    /// Returns a struct containing image metadata.
+    ///
+    /// If the image is stored correctly, the function tries to fetch the image from the storage.
+    ///
+    /// If the image is not stored or a storage was not created (the 'storage' field is None),
+    /// it pulls the image, stores it (if the 'storage' field is not None) and returns its metadata.
+    ///
+    /// If the pull succeeded but the store operation failed, it returns the pulled image metadata.
+    async fn get_image_details(&mut self, image_name: &str) -> Result<ImageDetails> {
+        let image_name = check_tag(image_name);
+
+        let local_storage = (self.storage).as_mut();
+
+        if let Some(storage) = local_storage {
+            // Try to fetch the image from the storage
+            let image_details = storage.fetch_image_details(&image_name).map_err(|err| {
+                // Log the fetching error
+                eprintln!("{:?}", err);
+                err
+            });
+
+            // If the fetching failed, pull it from remote and store it
+            return image_details.or(self.fetch_and_try_store_image(&image_name).await);
+        }
+
+        self.fetch_and_try_store_image(&image_name).await
+    }
+
+    /// Pulls image from remote registry and stores it if possible
+    async fn fetch_and_try_store_image(&mut self, image_name: &str) -> Result<ImageDetails> {
+        // The image is not stored, so try to pull and then store it
+        let image_data = crate::pull::pull_image_data(image_name).await?;
+
+        // If the store operation failed, still return the image details
+        if let Some(local_storage) = self.storage.as_mut() {
+            local_storage
+                .store_image_data(image_name, &image_data)
+                .map_err(|err| eprintln!("Failed to store image: {:?}", err))
+                .ok();
+        }
+
+        // Get the image metadata from the pulled struct
+        let image_details = ImageDetails::build_details(image_name, &image_data)?;
+
+        Ok(image_details)
+    }
+
+    /// Extracts from the image and returns the CMD and ENV expressions (in this order).
+    ///
+    /// If there are no CMD expressions found, it tries to locate the ENTRYPOINT command.
+    fn extract_image(&mut self) -> Result<(Vec<String>, Vec<String>)> {
```

Comment:
Same comment here: the method should take an immutable reference judging by its name.